clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -51,6 +52,11 @@ private RecordMetadata(TopicPartition topicPartition, long offset, long timestam
         this.topicPartition = topicPartition;
     }
 
+    @Deprecated
+    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset) {
+        this(topicPartition, baseOffset, relativeOffset, Record.NO_TIMESTAMP, -1, -1, -1);
```

Comment:
A quick search in Github shows a couple places in tests where people use the RecordMetadata constructor (old and new):
- https://github.com/jasperavisser/octowight/blob/bc8b58ab8f3c6f7f860f124e64e010c2d8cc92b7/event-emitter/src/test/scala/nl/haploid/octowight/emitter/service/EventChannelServiceTest.scala#L35
- https://github.com/akka/reactive-kafka/blob/693d4a9d254357e7e3ab98401633b6b3c018a78b/core/src/test/scala/akka/kafka/internal/ProducerTest.scala
- https://github.com/nokia/wookie/blob/76cf9537a3ad835f098c7450ebce918abd35e71f/collector-api/src/test/scala/wookie/collector/cli/KafkaPusherAppSpec.scala
- https://github.com/confluentinc/kafka-rest/blob/b071562041f45ff738f39faff659aebf8eaa8e61/src/test/java/io/confluent/kafkarest/unit/TopicsResourceAvroProduceTest.java#L96